diffrax/_solver/leapfrog_midpoint.py

```diff
@@ -75,12 +85,36 @@ def step(
         made_jump: BoolScalarLike,
     ) -> tuple[Y, _ErrorEstimate, DenseInfo, _SolverState, RESULTS]:
         del made_jump
-        tm1, ym1 = solver_state
+        tm1, ym1, dt = solver_state
         control = terms.contr(tm1, t1)
         y1 = (ym1**ω + terms.vf_prod(t0, y0, args, control) ** ω).ω
         dense_info = dict(y0=y0, y1=y1)
-        solver_state = (t0, y0)
+        solver_state = (t0, y0, dt)
         return y1, None, dense_info, solver_state, RESULTS.successful
 
+    def backward_step(
+        self,
+        terms: AbstractTerm,
+        t0: RealScalarLike,
+        t1: RealScalarLike,
+        y1: Y,
+        args: Args,
+        solver_state: _SolverState,
+        made_jump: BoolScalarLike,
+    ) -> tuple[Y, DenseInfo, _SolverState, RESULTS]:
+        del made_jump
+        t0, y0, dt = solver_state
+        tm1 = t0 - dt
+        control = terms.contr(tm1, t1)
+        ym1 = (y1**ω - terms.vf_prod(t0, y0, args, control) ** ω).ω
+        dense_info = dict(y0=y0, y1=y1)
+        # On the last step we need to make sure our solver state is correct
+        # (i.e. the state used on the forward). Otherwise, in `ReversibleAdjoint`,
+        # we would take a local forward step from an incorrect `solver_state`.
+        solver_state = jax.lax.cond(
+            tm1 > 0, lambda _: (tm1, ym1, dt), lambda _: (t0, y0, dt), None
```

Comment:
I think this predicate is assuming we're solving over the time interval of the form `[0, T]`? But in practice we might have that the left endpoint is nonzero.

This aside it's (a) possible to use `jnp.where`, which is lower-overhead for small stuff like this, but also (b) I think `lax.cond(pred, lambda: foo, lambda: bar)` should work, without arguments.